tools/nnyeah/nnyeah/Program.cs

```diff
@@ -35,22 +57,20 @@ static void Main (string [] args)
 			}
 
 			if (infile is null || outfile is null) {
-				Console.Error.WriteLine (Errors.E0014);
-				Environment.Exit (1);
+				throw new ConversionException (Errors.E0014);
 			}
 
 			// TODO - Long term this should default to files packaged within the tool but allow overrides
 			if (xamarinAssembly is null || microsoftAssembly is null) {
-				Console.Error.WriteLine (Errors.E0015);
-				Environment.Exit (1);
+				throw new ConversionException (Errors.E0015);
 			}
 
 			if (doHelp) {
 				PrintOptions (options, Console.Out);
-				Environment.Exit (0);
 			}
-			ProcessAssembly (xamarinAssembly!, microsoftAssembly!, infile!,
-				outfile!, verbose, forceOverwrite, suppressWarnings);
+			else {
+				ProcessAssembly (xamarinAssembly!, microsoftAssembly!, infile!, outfile!, verbose, forceOverwrite, suppressWarnings);
+			}
```

Comment:
If you changed this method to return the exit code, you won't need this change, you can just `return 0;` after `PrintOptions`.

Reply:
I had that before. I felt that showing the flow as explicit "if show help do that, else do all of the processing" was better than an early return. 

Thoughts?

Reply:
Yeah, it's more of a style choice, either way is fine with me.